hummingbot/connector/exchange/bybit/bybit_exchange.py

```diff
@@ -0,0 +1,887 @@
+                update_timestamp=self.current_timestamp,
+                new_state=OrderState.FAILED,
+            )
+            self._order_tracker.process_order_update(order_update)
+
+    async def _execute_cancel(self, client_order_id: str) -> Dict[str, Any]:
+        """
+        Requests the exchange to cancel an active order
+        :param client_order_id: the client id of the order to cancel
+        """
+        tracked_order = self._order_tracker.fetch_tracked_order(client_order_id)
+        if tracked_order is not None:
+            try:
+                api_params = {
+                    "orderLinkId": client_order_id,
+                }
+                cancel_result = await self._api_request(
+                    method=RESTMethod.DELETE,
+                    path_url=CONSTANTS.ORDER_PATH_URL,
+                    params=api_params,
+                    is_auth_required=True)
+
+                order_update: OrderUpdate = OrderUpdate(
+                    client_order_id=client_order_id,
+                    trading_pair=tracked_order.trading_pair,
+                    update_timestamp=self.current_timestamp,
+                    new_state=OrderState.CANCELED,
+                )
+                self._order_tracker.process_order_update(order_update)
+                return cancel_result
+
+            except asyncio.CancelledError:
+                raise
+            except Exception:
+                self.logger().network(f"There was a an error when requesting cancellation of order {client_order_id}")
+                raise
+
+    async def _status_polling_loop(self):
+        """
+        Performs all required operation to keep the connector updated and synchronized with the exchange.
+        It contains the backup logic to update status using API requests in case the main update source (the user stream
+        data source websocket) fails.
+        It also updates the time synchronizer. This is necessary because Bybit require the time of the client to be
+        the same as the time in the exchange.
+        Executes when the _poll_notifier event is enabled by the `tick` function.
+        """
+        while True:
+            try:
+                await self._poll_notifier.wait()
+                await self._update_time_synchronizer()
+                await safe_gather(self._update_balances(),
+                                  self._update_order_status())
+                self._last_poll_timestamp = self.current_timestamp
+
+                self._poll_notifier = asyncio.Event()
+            except asyncio.CancelledError:
+                raise
+            except Exception:
+                self.logger().network("Unexpected error while fetching account updates.", exc_info=True,
+                                      app_warning_msg="Could not fetch account updates from Bybit. "
+                                                      "Check API key and network connection.")
+                await asyncio.sleep(0.5)
```

Comment:
If the logic calls `asyncio.sleep` directly the unit tests will not be able to avoid the sleep during their execution.